sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -224,29 +185,56 @@ function processFormData(formData?: FormDataMap) {
 
 /**
  * Prepares the response body
+ * @param response - The received response
+ * @param asStream - The type to coerce the body as
  */
-function getResponseBody(response: PipelineResponse): RequestBodyType | undefined {
+async function getResponseBody(
+  response: PipelineResponse,
+  asStream: boolean = false
+): Promise<ResponseBodyType | undefined> {
   // Set the default response type
   const contentType = response.headers.get("content-type") ?? "";
   const firstType = contentType.split(";")[0];
-  const bodyToParse: string = response.bodyAsText ?? "";
+  const text = response.bodyAsText;
+  const stream = response.readableStreamBody ?? response.browserStreamBody;
+
+  if (
+    /**
+     * If the client is asking for a stream, return it directly.
+     */
+    asStream ||
+    /**
+     * If the response contains a stream, return it directly.
+     */
+    stream !== undefined
+  ) {
+    return stream;
+  }
 
   if (firstType === "text/plain") {
-    return String(bodyToParse);
+    return String(text);
```

Comment:
This seems to changed previous behavior when `bodyAsText` is `undefined`

Reply:
should `text/plain` case be moved to above to keep original behavior, even `stream` is not undefined?

Reply:
You're right, if the body is undefined, the string created will be `"undefined"` which is undesirable. I am not sure if this would be a valid body to begin with but you never know. Addressed in https://github.com/Azure/azure-sdk-for-js/pull/27000/commits/2e0539d6aee9e24f5e98da1915dcc0deed7f3a58.

Regarding the stream taking precedence, I think generally speaking a stream is favorable to a string. Notice that currently a stream will be returned if it was explicitly requested or if it was for SSEs.